backend/src/apiserver/server/pipeline_server.go

```diff
@@ -592,6 +601,23 @@ func (s *PipelineServer) validatePipelineVersionBeforeCreating(p *model.Pipeline
 	return util.NewInvalidInputError("Pipeline version must have a pipeline spec or a valid source code's URL. PipelineSpec: %s. PipelineSpecURI: %s. CodeSourceUrl: %s. At least one of them must have a valid pipeline spec", p.PipelineSpec, p.PipelineSpecURI, p.CodeSourceUrl)
 }
 
+// Validates a pipeline name to match the requirements. This function matches the constraints in the frontend.
+func validatePipelineName(pipelineName string) error {
+	if pipelineName == "" {
+		return util.NewInvalidInputError("pipeline's name cannot be empty")
+	}
+	if len(pipelineName) > 100 {
+		return util.NewInvalidInputError("pipeline's name must contain no more than 100 characters")
+	}
+	if matched, err := regexp.MatchString(pipelineNamePattern, pipelineName); err != nil {
+		return util.NewInternalServerError(
```

Comment:
This should never happen, our unit test can already proof it has no problem compiling the regex.

Reply:
True, this error is very unlikely. I still believe having this error message gives more options for troubleshooting issues caused by factors that out of our control like future changes or bugs in `regexp` package or sporadic runtime issues.

Reply:
> future changes or bugs in `regexp` package 

This if happens would be captured by our presubmit.

> or sporadic runtime issues.

For the natural of what it does, it would be very unlikely to have flakiness

I still prefer cleaner code to the little chance of this code ever providing any value, but no strong opinion.